### PR TITLE
feat(tiproxy): add tiproxy connector test

### PR DESCRIPTION
**Pull request by xhebox**

### CODE REVIEW

File: pipelines/pingcap/tiproxy/latest/pull_mysql_test.groovy [hunk 1]
```diff
@@ -83,41 +93,58 @@ pipeline {
...
+                  }
+              }
+          }
+          stage('MySQL Connector Tests') {
+              steps {
+                  dir('tidb-test') {
+                      cache(path: "./", filter: '**/*', key: "ws/${BUILD_TAG}/tiproxy-mysql-connector-test") {
+                          sh label: "PART ${PART}", script: """
+                              #!/usr/bin/env bash
+															./bin/tidb-server &
```
Comment (purelind) [inline pipelines/pingcap/tiproxy/latest/pull_mysql_test.groovy:139]: Please adjust the indentation.

Comment (xhebox) [reply pipelines/pingcap/tiproxy/latest/pull_mysql_test.groovy]: Done.

File: pipelines/pingcap/tiproxy/latest/pull_mysql_test.groovy [hunk 2]
```diff
@@ -59,6 +59,16 @@ pipeline {
                         }
                     }
                 }
+                dir("mysql-server") {
+                    cache(path: "./", filter: '**/*', key: "git/xhebox/mysql-server/rev-${REFS.pulls[0].sha}", restoreKeys: ['git/xhebox/mysql-server/rev-']) {
+                        retry(2) {
+                            script {
+                                component.checkout('git@github.com:xhebox/mysql-server.git', 'mysql-server', "8.0", REFS.pulls[0].title, GIT_CREDENTIALS_ID)
```
Comment (purelind) [inline pipelines/pingcap/tiproxy/latest/pull_mysql_test.groovy:66]:
```suggestion
                                component.checkout('https://github.com/xhebox/mysql-server.git', 'mysql-server', "8.0", REFS.pulls[0].title, GIT_CREDENTIALS_ID)
```

File: pipelines/pingcap/tiproxy/latest/pull_mysql_test.groovy [hunk 3]
```diff
@@ -83,41 +93,58 @@ pipeline {
...
+              }
+          }
+          stage('MySQL Connector Tests') {
+              steps {
+                  dir('tidb-test') {
+                      cache(path: "./", filter: '**/*', key: "ws/${BUILD_TAG}/tiproxy-mysql-connector-test") {
+                          sh label: "PART ${PART}", script: """
+                              #!/usr/bin/env bash
+                              ./bin/tidb-server &
+                              TIDB_PID=$!
```
Comment (purelind) [inline pipelines/pingcap/tiproxy/latest/pull_mysql_test.groovy:140]:
```suggestion
                              TIDB_PID=\$!
```

File: pipelines/pingcap/tiproxy/latest/pull_mysql_test.groovy [hunk 4]
```diff
@@ -83,41 +93,58 @@ pipeline {
...
+                              ./mysql_client_test/test.sh -l 127.0.0.1 -p 4000 -t $PWD/../tiproxy -m $PWD/../mysql-server -u root
```
Comment (purelind) [inline pipelines/pingcap/tiproxy/latest/pull_mysql_test.groovy:141]:
```suggestion
                              ./mysql_client_test/test.sh -l 127.0.0.1 -p 4000 -t \$PWD/../tiproxy -m \$PWD/../mysql-server -u root
```

File: pipelines/pingcap/tiproxy/latest/pull_mysql_test.groovy [hunk 5]
```diff
@@ -83,41 +93,58 @@ pipeline {
...
+                              kill $TIDB_PID || true
```
Comment (purelind) [inline pipelines/pingcap/tiproxy/latest/pull_mysql_test.groovy:142]:
```suggestion
                              kill \$TIDB_PID || true
```

File: pipelines/pingcap/tiproxy/latest/pull_mysql_test.groovy [hunk 6]
```diff
@@ -59,6 +59,16 @@ pipeline {
...
+                dir("mysql-server") {
+                    cache(path: "./", filter: '**/*', key: "git/xhebox/mysql-server/rev-${REFS.pulls[0].sha}", restoreKeys: ['git/xhebox/mysql-server/rev-']) {
+                        retry(2) {
+                            script {
+                                component.checkout('https://github.com/xhebox/mysql-server.git', 'mysql-server', "8.0", REFS.pulls[0].title, GIT_CREDENTIALS_ID)
+                            }
+                        }
+                    }
+                }
+            }
```
Comment (purelind) [inline pipelines/pingcap/tiproxy/latest/pull_mysql_test.groovy:71]: remove this extra curly brackets

Comment (xhebox) [reply pipelines/pingcap/tiproxy/latest/pull_mysql_test.groovy:71]: Done

**Comment by purelind** [comment]
https://github.com/PingCAP-QE/ci/pull/2274

**Pull request closed without merging**